File: 204_Count_Primes.js
var countPrimes = function(n) {
    if(n === 0 || n === 1 || n === 2) return 0;
    let count = 0;
    let i = 2;
    while(i<n){
        let j = 2;
        let flag = 0;
        while(j<i){
            if(i%j === 0){
                flag = 1;
                break;
            }
            j++;
        }
        if(flag === 0) count++;
        i++;
    }
    return count;
};

File: 204_Count_Primes.cpp
class Solution {
public:
    int countPrimes(int n) {
        vector<int> prime(n,0);
        for(long long i = 2; i <= n; i++){
            long long j = i * i;
            for(; j < n; j += i){
                prime[j] = 1;
            }
        }
        int count = 0;
        for(int i = 2; i < n; i++){
            if(prime[i] == 0) {
                count++;
            }
        }
        return count;
    }
};
